api/swagger.yml

```
@@ -421,6 +421,28 @@ components:
           type: boolean
           default: false
 
+    DiffObjectInfo:
```

Can we use ObjectStats?  It seems like it should be that same type.  (An advantage of using the same type is that you often need less code in the client.  Of course only if it really is the same object, not just another object with the same fields.  Here AFAICT these _are_ the object stats after the change - and in future before it, too.)

But even if not, this object has nothing to do with "diffs".  So perhaps
```suggestion
    ObjectInfo:
```

---

I mean, we can use ObjectStats but we will always return several fields empty such as the `path` and `physical_path`, as we don't want to return them in this context...
We probably should use a specialized struct for the diff methods after all...